template/src/VideoSample/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {useCallback, useEffect, useMemo} from 'react';
+import * as RN from 'react-native';
+import {Types, Fonts, Theme} from '@sleeperhq/mini-core';
+import {FlashList} from '@shopify/flash-list';
+import {Topic} from '@sleeperhq/mini-core/declarations/types';
+import VideoRow from './VideoRow';
+
+type VideoMiniProps = {
+  context: Types.Context;
+};
+
+const PAGE_LIMIT = 20;
+const EMPTY_OBJECT = {};
+
+const Videos = (props: VideoMiniProps) => {
+  const {context} = props;
+
+  const [page, setPage] = React.useState(0);
+  const [isRefreshing, setIsRefreshing] = React.useState(false);
+  const [paginatedVideosMap, setPaginatedVideoMap] =
+    React.useState<Record<string, Topic[]>>(EMPTY_OBJECT);
+
+  const currentPageVideos = context?.videos?.[page] || [];
```

Comment:
One thing to check - does this need to be a `useRef()`? I'm wondering if the `useEffect` below will ever have stale data otherwise.

Reply:
I don't think so, at least it seems to be working without issue.